### PR TITLE
Add `<meta>` tags to template

### LINKED ISSUE #820: Dropdown button for thumbnail script.
I noticed that many users don't know how to set the thumbnail in their brews. Perhaps this can be one of the options in the dropdown menus?

`<meta property="og:image" content="LINKTOIMAGE" />`

### PR DESCRIPTION
**Pull request by G-Ambatte**
This PR adds `<meta>` tags to the template function.

- `og:title` = brew.title OR "Homebrewery - Untitled Brew"
- `og:type` = 'website'
- `og:description` = brew.description OR 'No description.'
- `og:url` = brew.shareId OR 'homebrewery.naturalcrit.com/new'
- `og:image` = brew.thumbnail OR 'https://i.imgur.com/FwRuhv7.png'
- `og:site_name` = 'The Homebrewery - Make your Homebrew content look legit!'

This will resolve #820 and may go some way towards resolving #294 .

### CODE REVIEW

File: client/homebrew/editor/metadataEditor/metadataEditor.jsx [hunk 1]
```diff
@@ -11,19 +11,27 @@ const MetadataEditor = createClass({
...
+				thumbnail     : '',
+				showThumbnail : false
 			},
 			onChange : ()=>{}
 		};
 	},
 
+	getInitialState : function(){
+		return {
+			showThumbnail : false
```
[the remark this answers is not in the document]
Comment (G-Ambatte) [reply client/homebrew/editor/metadataEditor/metadataEditor.jsx]: Default changed to visible.

File: client/homebrew/editor/metadataEditor/metadataEditor.jsx [hunk 2]
```diff
@@ -65,6 +73,17 @@ const MetadataEditor = createClass({
...
 
+	toggleThumbnailDisplay : function(){
+		this.setState({
+			showThumbnail : !this.state.showThumbnail
+		});
+	},
+
+	renderThumbnail : function(){
+		if(!this.state.showThumbnail) return;
+		return <img className='thumbnail-preview' src={this.props.metadata.thumbnail || 'https://i.imgur.com/FwRuhv7.png'}></img>;
```
[the remark this answers is not in the document]
Comment (G-Ambatte) [reply client/homebrew/editor/metadataEditor/metadataEditor.jsx]: I've been using an Imgur icon during my initial development, it should definitely be changed to a locally hosted asset before we push live.

File: client/homebrew/editor/metadataEditor/metadataEditor.jsx [hunk 3]
```diff
@@ -161,6 +180,16 @@ const MetadataEditor = createClass({
 				<textarea value={this.props.metadata.description} className='value'
 					onChange={(e)=>this.handleFieldChange('description', e)} />
 			</div>
+			<div className='field thumbnail'>
+				<label>thumbnail URL</label>
```
[the remark this answers is not in the document]
Comment (G-Ambatte) [reply client/homebrew/editor/metadataEditor/metadataEditor.jsx]: Label changed and placeholder added to Thumbnail text input.

File: client/homebrew/editor/metadataEditor/metadataEditor.less [hunk 4]
```diff
@@ -24,6 +24,29 @@
...
+				color: white;
+				padding: 0px 5px;
+				background-color: black;
+				&:hover{
+					background-color: #777;
+				}
+			}
+			.thumbnail-preview{
+				position    : relative;
+				width       : 80px;
```
[the remark this answers is not in the document]
Comment (G-Ambatte) [reply client/homebrew/editor/metadataEditor/metadataEditor.less:48]: `max-height` added; I used 122px based on testing from my local install.

File: client/homebrew/pages/editPage/editPage.jsx [hunk 5]
```diff
@@ -45,7 +45,8 @@ const EditPage = createClass({
 				published   : false,
 				authors     : [],
 				systems     : [],
-				renderer    : 'legacy'
+				renderer    : 'legacy',
+				thumbnail   : ''
```
[the remark this answers is not in the document]
Comment (G-Ambatte) [reply client/homebrew/pages/editPage/editPage.jsx]: Actually, I don't think we need this at all... Testing on the local install with and without the `thumbnail` defined in default props has shown no functional differences.

File: client/homebrew/editor/metadataEditor/metadataEditor.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (G-Ambatte) [reply client/homebrew/editor/metadataEditor/metadataEditor.jsx]: Latest commit uses a locally hosted thumbnail file.

File: client/homebrew/editor/metadataEditor/metadataEditor.jsx [hunk 6]
```diff
@@ -161,6 +180,16 @@ const MetadataEditor = createClass({
...
+				<textarea value={this.props.metadata.thumbnail} className='value'
```
[the remark this answers is not in the document]
Comment (G-Ambatte) [reply client/homebrew/editor/metadataEditor/metadataEditor.jsx]: Fixed in the latest commit.

**Comment by G-Ambatte** [comment]
Defaults in `/new`:
image


Custom image in `/edit`:
image

**Comment by G-Ambatte** [comment]
Possible future extensions:
- Meta tags for specific social media sites (e.g. Twitter, Facebook, etc). *Limited benefit, may become considerable additional work and require ongoing work as social media sites change/update meta tag requirements*
- Publication check? Exclude meta tags from unpublished brews. *Seems like additional work for no additional benefit*

---

Additional reading: https://css-tricks.com/essential-meta-tags-social-media

**Comment by calculuschild** [comment]
OK, this seems to be working and I don't have any other comments.

I think it's good to merge?

**Comment by calculuschild** [comment]
@G-Ambatte 

After merging this I immediately found a critical error, so feel free to create a new PR here since we can't reopen PRs directly it seems.

The error is this: Google Drive has a 124-byte limit on any property added to a document. Long image urls will cause the saving step to fail, which is actually pretty common. We're going to need an alternative way to add this to the Google document.

```
errors: [
    {
      domain: 'global',
      reason: 'propertyLengthLimitExceeded',
      message: 'Properties and app properties are limited to 124 bytes in UTF-8 encoding, counting both the key and the value.'
    }
  ]
```

**Comment by Gazook89** [comment]
perhaps an error message that basically just says "URL is too long; try hosting on Imgur" or similar?

Does the same happen if a brew description is too long?

**Comment by calculuschild** [comment]
If we implement it as a Google Drive property now, it will only make the transition to the codefence stickier because then we have to handle the case of Google brews that were saved to store the date in a property and then never updated to use the new method. We would have to handle both cases which is something we can avoid otherwise.

Descriptions didn't have this problem because Google drive coincidentally has a "description" field that can hold several kilobytes and we have been using that. A long enough description would run into the same issue theoretically however.

There aren't any other "big" fields we can use. We only get the main text, the description, and 30 short properties.

File: client/template.js [hunk 7]
```diff
@@ -7,6 +7,12 @@ module.exports = async(name, title = '', props = {})=>{
 		<link href="//fonts.googleapis.com/css?family=Open+Sans:400,300,600,700" rel="stylesheet" type="text/css" />
 		<link href=${`/${name}/bundle.css`} rel='stylesheet' />
 		<link rel="icon" href="/assets/homebrew/favicon.ico" type="image/x-icon" />
+		<meta property="og:title" content="${props.brew?.title ? props.brew.title : 'Homebrewery - Untitled Brew'}">
```
[the remark this answers is not in the document]
Comment (G-Ambatte) [reply client/template.js:11]: Experimenting with putting the `<meta>` tags in the body shows that they're not read by FB Share Debugger - only the `<meta>` tags in the `<head>` are read.

[the remark this answers is not in the document]
Comment (calculuschild) [reply client/template.js:11]: Ok. This will be fine then.

**Pull request merged**